### PR TITLE
Add volume-group methods

### PR DESCRIPTION
**Pull request by dhirendersingh19**
- Add volume-group methods & terraform attributes

### CODE REVIEW

**Review by yussufsh — CHANGES_REQUESTED** [review]
Check if error messages can be made more compact.

File: helpers/constants.go [hunk 1]
```diff
@@ -66,6 +66,11 @@ const (
 	PIStoragePoolName         = "pi_storage_pool_name"
 	PIReplicationEnabled      = "pi_replication_enabled"
 
+	// IBM PI Volume Group
+	PIVolumeGroupName                 = "pi_volume_group_name"
+	PIVolumeGroupsVolumeIds           = "pi_volume_ids"
+	PIVolumeGroupConsistencyGroupName = "pi_consistency_group_name"
+
```
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply helpers/constants.go:73]: removing it from power-go-client, will add in terraform repo

File: examples/volume-group/main.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	}
+	log.Printf("***************[7]****************** %+v\n", *getLiveDetails)
+
+	getRCRelationships, err := powerClient.GetVolumeGroupRemoteCopyRelationships(vgID)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Printf("***************[8]****************** %+v\n", *getRCRelationships)
+
+	// removing volumes before deleting
```
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply examples/volume-group/main.go:122]: yes, update vg api have both addvolumes & removeVolumes attribute. and vg cannot be deleted if it has volume in it, so before deleting vg we have to remove all volumes from vg

File: clients/instance/ibm-pi-volume-group.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	if resp == nil || resp.Payload == nil {
+		return nil, fmt.Errorf("failed to get live details of volume-group %s", id)
+	}
+	return resp.Payload, nil
+}
+
+// Performs action on Volume Group
+func (f *IBMPIVolumeGroupClient) VolumeGroupAction(id string, body *models.VolumeGroupAction) error {
+	params := p_cloud_volume_groups.NewPcloudVolumegroupsActionPostParams().
+		WithContext(f.ctx).WithTimeout(helpers.PIDeleteTimeOut).
```
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply clients/instance/ibm-pi-volume-group.go:149]: yes, will update time out accordingly

File: clients/instance/ibm-pi-volume-group.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+		WithCloudInstanceID(f.cloudInstanceID).WithVolumeGroupID(id).WithBody(body)
+	_, err := f.session.Power.PCloudVolumeGroups.PcloudVolumegroupsActionPost(params, f.session.AuthInfo(f.cloudInstanceID))
```
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply clients/instance/ibm-pi-volume-group.go:151]: the response return is of models.object, should we return this response from the method?

File: clients/instance/ibm-pi-volume-group.go [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+func (f *IBMPIVolumeGroupClient) GetDetails(id string) (*models.VolumeGroupDetails, error) {
+	params := p_cloud_volume_groups.NewPcloudVolumegroupsGetDetailsParams().
+		WithContext(f.ctx).WithTimeout(helpers.PIGetTimeOut).
+		WithCloudInstanceID(f.cloudInstanceID).WithVolumeGroupID(id)
+	resp, err := f.session.Power.PCloudVolumeGroups.PcloudVolumegroupsGetDetails(params, f.session.AuthInfo(f.cloudInstanceID))
+	if err != nil {
+		return nil, fmt.Errorf(errors.GetVolumeGroupDetailsOperationFailed, id, f.cloudInstanceID, err)
+	}
+	if resp == nil || resp.Payload == nil {
+		return nil, fmt.Errorf("failed to get volume-group %s", id)
```
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply clients/instance/ibm-pi-volume-group.go]: changes done

File: clients/instance/ibm-pi-volume-group.go [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+	if err != nil {
+		return fmt.Errorf(errors.DeleteVolumeGroupOperationFailed, id, err)
+	}
+	return nil
+}
+
+// Get live details of a Volume Group
+func (f *IBMPIVolumeGroupClient) GetVolumeGroupLiveDetails(id string) (*models.VolumeGroupStorageDetails, error) {
+	params := p_cloud_volume_groups.NewPcloudVolumegroupsStorageDetailsGetParams().
+		WithContext(f.ctx).WithTimeout(helpers.PIDeleteTimeOut).
```
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply clients/instance/ibm-pi-volume-group.go]: done

File: clients/instance/ibm-pi-volume-group.go [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+	if err != nil {
+		return fmt.Errorf(errors.VolumeGroupActionOperationFailed, id, f.cloudInstanceID, err)
+	}
+	return nil
+}
+
+// Get remote copy relationships of the volume belonging to volume group
+func (f *IBMPIVolumeGroupClient) GetVolumeGroupRemoteCopyRelationships(id string) (*models.VolumeGroupRemoteCopyRelationships, error) {
+	params := p_cloud_volume_groups.NewPcloudVolumegroupsRemoteCopyRelationshipsGetParams().
+		WithContext(f.ctx).WithTimeout(helpers.PIDeleteTimeOut).
```
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply clients/instance/ibm-pi-volume-group.go]: done

File: clients/instance/ibm-pi-volume-group.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply clients/instance/ibm-pi-volume-group.go]: returning model.Object & added check for nil response

File: clients/instance/ibm-pi-volume-group.go [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+		WithCloudInstanceID(f.cloudInstanceID).WithBody(body)
+	respOk, respPartial, err := f.session.Power.PCloudVolumeGroups.PcloudVolumegroupsPost(params, f.session.AuthInfo(f.cloudInstanceID))
+	if err != nil {
+		return nil, fmt.Errorf(errors.CreateVolumeGroupOperationFailed, f.cloudInstanceID, err)
+	}
+	if respOk != nil && respOk.Payload != nil {
+		return respOk.Payload, nil
+	}
+	if respPartial != nil && respPartial.Payload != nil {
+		return respOk.Payload, nil
```
[the remark this answers is not in the document]
Comment (dhirendersingh19) [reply clients/instance/ibm-pi-volume-group.go]: done

**Review by yussufsh — APPROVED** [review]
lgtm

**Pull request merged**